### PR TITLE
Added coverage cmake script for nGraph

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Enabled coverage for nGraph components

### CODE REVIEW

File: ngraph/cmake/coverage.cmake [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(NGRAPH_COVERAGE_GCDA_DATA_DIRECTORY "${CMAKE_BINARY_DIR}/ngraph")
```
Comment (ilya-lavrenov) [inline ngraph/cmake/coverage.cmake:5]: let's have common directory for GCDA data for the whole OpenVINO

File: ngraph/cmake/coverage.cmake [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+set(NGRAPH_COVERAGE_BASE_DIRECTORY "${OpenVINO_MAIN_SOURCE_DIR}/ngraph")
```
Comment (ilya-lavrenov) [inline ngraph/cmake/coverage.cmake:6]: -> `${OpenVINO_MAIN_SOURCE_DIR}`, but update patterns

Comment (ilyachur) [reply ngraph/cmake/coverage.cmake]: Fixed

File: ngraph/cmake/coverage.cmake (hunk 1 above)
Comment (ilyachur) [reply ngraph/cmake/coverage.cmake]: Fixed

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**